### PR TITLE
fix: don't show banner if preferences exist

### PR DESCRIPTION
**Pull request by iamacook**
## What it solves

Resolves https://github.com/safe-global/safe-wallet-web/pull/2592#issuecomment-1749912718

## How this PR fixes it

The notification registration banner now takes into account whether the current Safe is registered before being displayed.

## How to test it

1. Ensure a Safe is added and _not_ registered for notifications.
2. Open the Safe in question but don't dismiss the notification registration banner.
3. Open the notification settings for that Safe.
4. Register for notifications.
5. Observe the banner close.
6. Reopening that Safe should not show the banner until it is "unregistered" for notifications.

## Screenshots

notification banner

## Checklist
* [ ] I've tested the branch on mobile 📱
* [ ] I've documented how it affects the analytics (if at all) 📊
* [ ] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

File: src/components/settings/PushNotifications/PushNotificationsBanner/index.tsx [hunk 1]
```diff
@@ -95,13 +95,15 @@ export const PushNotificationsBanner = ({ children }: { children: ReactElement }
...
   const onboard = useOnboard()
 
+  const { getPreferences, getAllPreferences } = useNotificationPreferences()
   const { dismissPushNotificationBanner, isPushNotificationBannerDismissed } = useDismissPushNotificationsBanner()
 
   const isSafeAdded = !!addedSafesOnChain?.[safeAddress]
-  const shouldShowBanner = isNotificationsEnabled && !isPushNotificationBannerDismissed && isSafeAdded
+  const isSafeRegistered = getPreferences(safe.chainId, safeAddress)
+  const shouldShowBanner =
+    isNotificationsEnabled && !isPushNotificationBannerDismissed && isSafeAdded && !isSafeRegistered
```
Comment (katspaugh) [inline src/components/settings/PushNotifications/PushNotificationsBanner/index.tsx:104]: Could you rename `isNotificationsEnabled` to something like `isFeatureEnabled`? It's confusing wrt user enabling notifications/allowing push notifications.

**Review by katspaugh — APPROVED** [review]

File: src/components/settings/PushNotifications/PushNotificationsBanner/index.tsx [hunk 2]
```diff
@@ -87,21 +87,23 @@ export const _getSafesToRegister = (
...
   const addedSafesOnChain = useAppSelector((state) => selectAddedSafes(state, safe.chainId))
   const { query } = useRouter()
   const onboard = useOnboard()
 
+  const { getPreferences, getAllPreferences } = useNotificationPreferences()
   const { dismissPushNotificationBanner, isPushNotificationBannerDismissed } = useDismissPushNotificationsBanner()
 
   const isSafeAdded = !!addedSafesOnChain?.[safeAddress]
-  const shouldShowBanner = isNotificationsEnabled && !isPushNotificationBannerDismissed && isSafeAdded
+  const isSafeRegistered = getPreferences(safe.chainId, safeAddress)
```
Comment (schmanu) [inline src/components/settings/PushNotifications/PushNotificationsBanner/index.tsx:102]: Should we write a unittest for this component?

**Review by schmanu — APPROVED** [review]
Looks good to me. We could consider writing a test for the Banner component as it's complexity grew a bit.

File: src/components/settings/PushNotifications/PushNotificationsBanner/index.tsx (hunk 2 above)
Comment (iamacook) [reply src/components/settings/PushNotifications/PushNotificationsBanner/index.tsx:102]: I've added tests covering the logic related to this PR. I'll add another test in the other open one as well.

**Comment by francovenica** [comment]
Looks good.

The pop up dissapears if you enable notifications in the safe settings
The pop up doesn't show if you enable the notifications from the global settings and then visit the safe.

**Pull request merged**